### PR TITLE
[engraving] replaced QList to std. Part 4

### PR DESCRIPTION
**Pull request by igorkorsukov**
I tried to reduce the number of warnings, but there are still a lot of them, the  work continues...

### CODE REVIEW

**Review by handrok — APPROVED** [review]

File: src/engraving/libmscore/barline.cpp [hunk 1]
```diff
@@ -570,7 +570,7 @@ bool BarLine::isBottom() const
     if (!_spanStaff) {
         return true;
     }
-    int idx = staffIdx();
+    size_t idx = staffIdx();
     if (idx == score()->nstaves() - 1) {
```
Comment (Jojo-Schmitz) [inline src/engraving/libmscore/barline.cpp:574]: Here `idx` could potentially go negative, if `nstaves()` is 0, or is that something not to consider?

File: src/engraving/libmscore/bracketItem.cpp [hunk 2]
```diff
@@ -51,7 +51,7 @@ mu::engraving::PropertyValue BracketItem::getProperty(Pid id) const
     case Pid::BRACKET_COLUMN:
         return _column;
     case Pid::BRACKET_SPAN:
-        return _bracketSpan;
+        return int(_bracketSpan);
```
Comment (Jojo-Schmitz) [inline src/engraving/libmscore/bracketItem.cpp:54]: Hmm, not `static_cast<int>(_bracketSpan)`?

File: src/engraving/libmscore/bracketItem.cpp [hunk 3]
```diff
@@ -67,7 +67,7 @@ bool BracketItem::setProperty(Pid id, const engraving::PropertyValue& v)
         staff()->changeBracketColumn(column(), v.toInt());
         break;
     case Pid::BRACKET_SPAN:
-        _bracketSpan = v.toInt();
+        _bracketSpan = size_t(v.toInt());
```
Comment (Jojo-Schmitz) [inline src/engraving/libmscore/bracketItem.cpp:70]: similar

File: src/engraving/libmscore/barline.cpp (hunk 1 above)
Comment (igorkorsukov) [reply src/engraving/libmscore/barline.cpp:574]: I looked that there is an equality check, so no problem if there was -1

File: src/engraving/libmscore/bracketItem.cpp (hunk 2 above)
Comment (igorkorsukov) [reply src/engraving/libmscore/bracketItem.cpp]: fixed

File: src/engraving/libmscore/bracketItem.cpp (hunk 3 above)
Comment (igorkorsukov) [reply src/engraving/libmscore/bracketItem.cpp]: fixed

**Pull request merged**